hashes/src/internal_macros.rs

```diff
@@ -207,6 +206,24 @@ macro_rules! hash_type {
             fn internal_new(arr: [u8; $bits / 8]) -> Self { Hash(arr) }
 
             fn internal_engine() -> HashEngine { Default::default() }
+
+            /// Construts this wrapper type from an array.
+            ///
+            /// Note: this function works the same as the one in the `Hash` trait but is `const`.
+            #[inline]
+            pub const fn from_byte_array(bytes: [u8; $bits / 8]) -> Self { Hash(bytes) }
+
+            /// Returns the raw hash bytes.
+            ///
+            /// Note: this function works the same as the one in the `Hash` trait but is `const`.
+            #[inline]
+            pub const fn to_byte_array(self) -> [u8; $bits / 8] { self.0 }
+
+            /// Returns the reference to raw hash bytes.
```

Comment:
I'm not sure about the use of the term 'raw' when we have other functions that use raw in the identifier `from_raw_bytes` but I couldn't think of a better wording.

Reply:
"underlying" may be better.

Reply:
If we want to stick to more outdoorsy words, we could consider "bare" :)